compiler/semantic/sql.go

```diff
@@ -468,9 +534,16 @@ func (a *analyzer) deriveAs(as ast.Assignment) (field.Path, error) {
 	return nil, fmt.Errorf("AS clause not a field")
 }
 
-func (a *analyzer) sqlField(e ast.Expr) field.Path {
-	if f, ok := a.semField(e).(*dag.This); ok {
-		return f.Path
+func (a *analyzer) semGroupByKey(in ast.Expr) (*dag.This, bool) {
+	e := a.semExpr(in)
+	this, ok := e.(*dag.This)
+	if !ok {
+		a.error(in, errors.New("GROUP BY expressions are not yet supported; try expression in the selection with an AS"))
+		return nil, false
+	}
+	if len(this.Path) == 0 {
+		a.error(in, errors.New("cannot use 'this' as group-by expression"))
```

Comment:
Nit: Matching styling in other error strings.
```suggestion
		a.error(in, errors.New("cannot use 'this' as GROUP BY expression"))
```